chain/challenges-10.test.js

```diff
@@ -0,0 +1,198 @@
+
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named count that, given an integer and an array of arrays, uses either
+// filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+//
+// Note: You might need to use the same method more than once.
+//
+// For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+// ------------------------------------------------------------------------------------------------
+
+const count = (target, input) => {
+  // Solution code here...
+  let answer = 0;
+  input.map( ele => {
+      ele.map( integer => {
+          if(integer === target){
+              answer++;
+          }
+      })
+  })
+  return answer;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function that, given an array of integer arrays as input, either filter, map, or reduce
+// to calculate the total sum of all the elements in the array.
+//
+// Note: You might need to use the same method more than once.
+// ------------------------------------------------------------------------------------------------
+
+const totalSum = (input) => {
+  // Solution code here...
+  return input.reduce( (acc, ele) => acc + ele.reduce( (acc, ele) => acc+ele, 0), 0);
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+
+// Write a function named divisibleByFiveTwoToThePower that accepts an array of arrays as input.
+//
+// This function should first remove any elements that are not numbers or are not divisible by five.
+//
+// This function should then raise 2 to the power of the resulting numbers, returning an array of arrays.
+//
+// For example, [ [0,2,5,4], [2,4,10], [] ] should return [ [1, 32], [1024], [] ].
+// ------------------------------------------------------------------------------------------------
+
+const divisibleByFiveTwoToThePower = (input) => {
+  // Solution code here...
+  return input.map( arr => {
+      return arr.map( ele => {
```

Comment:
Doing things in the opposite order... interesting! As we mentioned in class, it probably makes more sense to filter first (based on the conditional in your map) and then do the map to raise 2 to that power.